sky/shell/platform/ios/accessibility_bridge.mm

```diff
@@ -0,0 +1,190 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/shell/platform/ios/accessibility_bridge.h"
+
+#include <UIKit/UIKit.h>
+
+namespace sky {
+namespace shell {
+namespace a11y {
+
+AccessibilityBridge::AccessibilityBridge(
+    FlutterView* view,
+    semantics::SemanticsServer* semanticsServer)
+    : view_(view), binding_(this), weak_factory_(this) {
+  mojo::InterfaceHandle<semantics::SemanticsListener> listener;
+  binding_.Bind(&listener);
+  semanticsServer->AddSemanticsListener(listener.Pass());
+}
+
+void AccessibilityBridge::UpdateSemanticsTree(
+    mojo::Array<semantics::SemanticsNodePtr> nodes) {
+  for (const semantics::SemanticsNodePtr& node : nodes) {
+    UpdateNode(node);
+  }
+
+  NSArray* accessibleElements = CreateAccessibleElements();
+  view_.accessibilityElements = accessibleElements;
+  [accessibleElements release];
+
+  UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification,
+                                  nil);
+}
+
+base::WeakPtr<AccessibilityBridge> AccessibilityBridge::AsWeakPtr() {
+  return weak_factory_.GetWeakPtr();
+}
+
+NodePtr AccessibilityBridge::UpdateNode(
+    const semantics::SemanticsNodePtr& node) {
+  NodePtr persistentNode;
+  NodeMap::iterator iter = nodes_.find(node->id);
+  if (iter == nodes_.end()) {
+    persistentNode = new Node(this, node);
+    nodes_.insert(NodeMap::value_type(node->id, persistentNode));
+  } else {
+    persistentNode = iter->second;
+    persistentNode->Update(node);
+  }
+  assert(persistentNode != nullptr);
+  return persistentNode;
+}
+
+void AccessibilityBridge::RemoveNode(NodePtr node) {
+  assert(nodes_.find(node->id_) != nodes_.end());
+  assert(nodes_.at(node->id_)->parent_ == nullptr);
+  nodes_.erase(node->id_);
+  for (NodePtr& child : node->children_) {
+    child->parent_ = nullptr;
+    RemoveNode(child);
+  }
+}
+
+NSArray* AccessibilityBridge::CreateAccessibleElements() const
+    NS_RETURNS_RETAINED {
+  NSMutableArray* accessibleElements = [[NSMutableArray alloc] init];
+  for (const auto& iter : nodes_) {
+    // TODO(tvolkert): n>1 roots will cause unstable sort here.
+    // Verify that there should only ever be one root, and perhaps
+    // store it so we don't have to search for it here.
+    if (iter.second->parent_ == nullptr) {
```

Comment:
Yeah.  I updated the TODO.  I'm working on a follow-on change to keep a tree of UIAccessibilityElements in which I'll make this change as well.